zenml/cli/pipeline.py

```diff
@@ -31,3 +37,106 @@ def set_metadata_store():
     click.echo('Comparing pipelines in repo: Starting app..')
     repo: Repository = Repository.get_instance()
     repo.compare_pipelines()
+
+
+@pipeline.command('list')
+def list_pipelines():
+    """Lists pipelines in the current repository."""
+    try:
+        repo: Repository = Repository.get_instance()
+    except Exception as e:
+        error(e)
+        return
+
+    pipelines = repo.get_pipelines()
+
+    names = [p.name for p in pipelines]
+    types = [p.PIPELINE_TYPE for p in pipelines]
+    statuses = [p.get_status() for p in pipelines]
+    cache_enabled = [p.enable_cache for p in pipelines]
+    filenames = [p.file_name for p in pipelines]
+
+    headers = ["name", "type", "cache enabled", "status", "file name"]
+
+    click.echo(tabulate(zip(names, types, cache_enabled, statuses, filenames),
+                        headers=headers))
+
+
+@pipeline.command('get')
+@click.argument('pipeline_id')
+def get_pipeline_by_id(pipeline_id: Text):
```

Comment:
We are not using ID's as a user-facing concept. Rather, the `names` of pipelines are unique and there is a built in function `get_pipeline_by_name` in repo to handle that,

In general, any functionality which is native to `fetching` pipelines should be a layer deeper within the `Repository` class, rather than the CLI layer :-) CLI should just be a wrapper that calls these functions and formats the output in a CLI manner

Reply:
Updated logic to get a pipeline by name instead, see commit f1bc3bb.